src/main/java/org/fundacionjala/virtualassistant/player/spotify/client/SpotifyClient.java

```diff
@@ -0,0 +1,364 @@
+                HttpMethod.GET,
+                request,
+                String.class
+        );
+
+        return response.getBody();
+    }
+
+    @Override
+    public String getSavedTracks(String accessToken) {
+        RestTemplate restTemplate = new RestTemplate();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + accessToken);
+
+        HttpEntity<String> request = new HttpEntity<>(headers);
+
+        ResponseEntity<String> response = restTemplate.exchange(
+                "https://api.spotify.com/v1/me/tracks",
+                HttpMethod.GET,
+                request,
+                String.class
+        );
+
+        return response.getBody();
+    }
+
+    @Override
+    public String getFollowed(String accessToken) {
+        RestTemplate restTemplate = new RestTemplate();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + accessToken);
+
+        HttpEntity<String> request = new HttpEntity<>(headers);
+
+        ResponseEntity<String> response = restTemplate.exchange(
+                "https://api.spotify.com/v1/me/following?type=artist",
+                HttpMethod.GET,
+                request,
+                String.class
+        );
+
+        return response.getBody();
+    }
+
+    public String extractCurrentTrackUri(String playerData) {
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode jsonNode = objectMapper.readTree(playerData);
+
+            if (jsonNode.has("item")) {
+                JsonNode item = jsonNode.get("item");
+                if (item.has("uri")) {
+                    return item.get("uri").asText();
+                }
+            }
+
+            return null;
+        } catch (IOException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    @Override
+    public String getPlayerInfo(String accessToken) {
+        RestTemplate restTemplate = new RestTemplate();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + accessToken);
+
+        HttpEntity<String> request = new HttpEntity<>(headers);
+
+        ResponseEntity<String> response = restTemplate.exchange(
+                "https://api.spotify.com/v1/me/player",
+                HttpMethod.GET,
+                request,
+                String.class
+        );
+
+        return response.getBody();
+    }
+
+    public String extractPlayerData(String playerData) {
+        ObjectMapper objectMapper = new ObjectMapper();
+        StringBuilder result = new StringBuilder();
+
+        try {
+            JsonNode root = objectMapper.readTree(playerData);
+
+            String playbackType = root.path("currently_playing_type").asText();
+
+            result.append("Playback Type: ").append(playbackType).append("\n");
+
+            if (playbackType.equals("track")) {
```

Comment:
when comparing string please invert the comparison to have the known variable in the right side, so we avoid having null pointer exceptions. `"TRACK.equals(playbackType)"`